n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Following code is the implementation of the N-queen problem.
+ * The solution is to place the queens on board such that no queen is under attack by another queen.
+ * There can be a number of possible solutions for a given length of board.
+ * This implementation prints only one valid solution, it can be extended to print all possible valid solutions.
+ */
+
+import java.util.Arrays;
+
+public class NQueenProblem {
+
+    private static boolean isQueenSafe(int[][] board, int row, int column) {
+        int i;
+        int j;
```

Comment:
I tried reducing it but Codacy is showing error to split declaration in multiple lines.
Please refer the errors mentioned by the Codacy bot above.